### DISCUSSION TITLE
timeout-minutes being ignored?

### DISCUSSION THREAD

**Discussion by chingc** [community/community#27177] [Actions]
<p>I’ve noticed more than expected consumption of my organization’s GitHub Actions monthly minutes for the past few days.  Today I found what I believe to be the cause.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8ee4e745b2d5b54ddfc23e0cec3e9fa1380b9a05.png" data-download-href="/uploads/short-url/ko6dTpp38B156LSJJJbSpxbMKwZ.png?dl=1" title="Image%202020-03-13%20at%2017.29.27" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/471856/181102162-f201447d-1c90-42eb-a2e1-9c61dc70c173.png" alt="Image%202020-03-13%20at%2017.29.27" data-base62-sha1="ko6dTpp38B156LSJJJbSpxbMKwZ" width="690" height="205" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8ee4e745b2d5b54ddfc23e0cec3e9fa1380b9a05_2_690x205.png, https://aws1.discourse-cdn.com/github/original/2X/8/8ee4e745b2d5b54ddfc23e0cec3e9fa1380b9a05.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/8ee4e745b2d5b54ddfc23e0cec3e9fa1380b9a05.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8ee4e745b2d5b54ddfc23e0cec3e9fa1380b9a05_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Image%202020-03-13%20at%2017.29.27</span><span class="informations">958×286 27.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>We don’t explicitly set <code>timeout-minutes</code> in our config, but it defaults to 360 minutes according to the <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes" rel="nofollow noopener">documentation</a>.  Why did it take 72 hours for GitHub Actions to terminate this build?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/chingc">@chingc</a> ，</p>
<p>There are few notes need you pay attention to:</p>
<ol>
<li>
<p>When using GitHub-hosted runner to run jobs in the workflow, each jobs can run for up to 6 hours （360 minutes） of execution time, , but this limit does not apply to self-hosted runners.</p>
</li>
<li>
<p>Each workflow run is limited to 72 hours. If a workflow run reaches this limit, the workflow run is cancelled. This limit also applies to self-hosted runners.</p>
</li>
</ol>
<p>According above two points, the below situations (including but not limited to) will let the workflow run be matked as timed out after 72 hours:</p>
<ol>
<li>
<p>When using GitHub-hosted runners to run some performed sequentially jobs (at least 12) in the workflow, and the total run time of the jobs reaches 72 hours.</p>
</li>
<li>
<p>When using self-hosted runners to run jobs (at least 1) in the workflow, one of the job run is stuck and finally make the run time of the workflow run reaches 72 hours.</p>
</li>
</ol>
<p>More details about the limits of time-out, you can reference here: <a href="https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits" rel="nofollow noopener">https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits</a></p>

**Comment by dskvr** [comment]
<p>I just got the error:</p>
<p><code>The job running on runner runner-1 has exceeded the maximum execution time of 360. </code></p>
<p>On a self-hosted runner.</p>
<p>You actually need to set `jobs.JOB_UID.minutes" to change the default 360 minute timeout.</p>
<p>More information here: <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes</a></p>
